client/state/notices/types.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+export type NoticeId = string;
+
+export type NoticeStatusError = 'is-error';
+export type NoticeStatusInfo = 'is-info';
+export type NoticeStatusPlain = 'is-plain';
+export type NoticeStatusSuccess = 'is-success';
+export type NoticeStatusWarning = 'is-warning';
+export type NoticeStatus =
+	| NoticeStatusError
+	| NoticeStatusInfo
+	| NoticeStatusPlain
+	| NoticeStatusSuccess
+	| NoticeStatusWarning;
+
+export type NoticeText = string | ReactNode;
+
+// Notice options that are always the same
+export interface BaseNoticeOptions {
+	button?: string;
+	displayOnNextPage?: boolean;
+	duration?: null | number;
+	href?: string;
+	isPersistent?: boolean;
+	onClick?: () => void;
+	showDismiss?: boolean;
+}
+
+// Notice options as they're expected by the action creator
+export interface NoticeOptions extends BaseNoticeOptions {
+	id?: NoticeId;
+}
+
+// Notice options as they're returned by the action creator
+export interface NoticeActionOptions extends BaseNoticeOptions {
+	noticeId: NoticeId;
+	status: NoticeStatus;
+	text: NoticeText;
+}
+
+// Notice action as returned by the action creator
+export type NoticeAction = {
+	type: string;
+	notice: NoticeActionOptions;
+};
+
+// Notice creation action creator function
+export type NoticeActionCreator = (
+	text: NoticeText,
+	noticeOptions?: NoticeOptions
+) => NoticeAction;
+
+// Higher level notice creation action creator function with status
+export type NoticeActionCreatorWithStatus = (
+	status: NoticeStatus,
+	text: NoticeText,
+	noticeOptions?: NoticeOptions
+) => NoticeAction;
+
+// Notice removal action creator function
+export type NoticeRemovalActionCreator = (
+	noticeId: NoticeId
+) => {
+	noticeId: NoticeId;
+	type: string;
```

Comment:
Could `type` be more specific here?

Reply:
Redux `type` is just a string anyway. We currently don't type our action types in any way. What would you suggest instead?

Reply:
I didn't realize we didn't type our actions at all. Nothing to do here in that case.

For what it's worth, typically the type of actions in Redux can by strictly typed as a union of types where they all share the same `type` property with strict values. An example can be found here: https://github.com/ItsJonQ/g2/blob/d8d4c53ad3eb6bbac50a61fe854dd47c088f4912/packages/components/src/Accordion/useAccordion.js#L77

Reply:
Actually, you're bringing a good point here. We don't need to allow all action types, but we only care about a couple of them. Fixed in c60d5d1089693b1724d1d722cee4c48d551a39e2